internal/cmd/kafka/command_link_create.go

```diff
@@ -167,36 +193,124 @@ func (c *linkCommand) create(cmd *cobra.Command, args []string) error {
 		if err != nil {
-			return nil, linkMode, err
+			return nil, nil, err
 		}
 		linkModeStr, ok := configMap["link.mode"]
 		if !ok {
 			// Default is destination if no config value is provided.
 			linkMode = Destination
+			linkModeStr = DESTINATION
 		} else if strings.EqualFold(linkModeStr, "DESTINATION") {
 			linkMode = Destination
+			linkModeStr = DESTINATION
 		} else if strings.EqualFold(linkModeStr, "SOURCE") {
 			linkMode = Source
+			linkModeStr = SOURCE
+		} else if strings.EqualFold(linkModeStr, "BIDIRECTIONAL") {
+			linkMode = Bidirectional
+			linkModeStr = BIDIRECTIONAL
 		} else {
-			return nil, linkMode, errors.Errorf(`unrecognized link.mode "%s". Use DESTINATION or SOURCE.`, linkModeStr)
+			return nil, &linkModeMetadata{linkMode, linkModeStr}, unrecognizedLinkModeErr(linkModeStr)
 		}
-		return configMap, linkMode, nil
+		configMap["link.mode"] = linkModeStr
+		return configMap, &linkModeMetadata{linkMode, linkModeStr}, nil
 	} else {
+		configMap := make(map[string]string)
+		configMap["link.mode"] = DESTINATION
 		// Default is destination if no config file is provided.
-		return make(map[string]string), Destination, nil
+		return configMap, &linkModeMetadata{Destination, DESTINATION}, nil
 	}
 }
 
-func (c *linkCommand) addSecurityConfigToMap(cmd *cobra.Command, linkMode linkMode, configMap map[string]string) error {
+func unrecognizedLinkModeErr(linkModeStr string) error {
+	return errors.Errorf(`unrecognized link.mode "%s". Use %s, %s or %s.`, linkModeStr, DESTINATION, SOURCE, BIDIRECTIONAL)
+}
+
+func (c *linkCommand) addSecurityConfigToMap(cmd *cobra.Command, linkModeMetadata *linkModeMetadata, configMap map[string]string) error {
+	linkMode := linkModeMetadata.linkMode
+	if linkMode == Source {
+		return c.addSourceInitiatedLinkSecurityConfigToMap(cmd, configMap)
+	} else if linkMode == Destination {
+		return c.addDestInitiatedLinkSecurityConfigToMap(cmd, configMap)
+	} else if linkMode == Bidirectional {
+		return c.addBidirectionalInitiatedLinkSecurityConfigToMap(cmd, configMap)
+	} else {
+		return unrecognizedLinkModeErr(linkModeMetadata.linkModeStr)
+	}
+}
+
+func (c *linkCommand) addBidirectionalInitiatedLinkSecurityConfigToMap(cmd *cobra.Command, configMap map[string]string) error {
+	remoteApiKey, err := cmd.Flags().GetString(remoteApiKeyFlagName)
+	if err != nil {
+		return err
+	}
+	remoteApiSecret, err := cmd.Flags().GetString(remoteApiSecretFlagName)
+	if err != nil {
+		return err
+	}
+	if remoteApiKey != "" && remoteApiSecret != "" {
+		configMap[securityProtocolPropertyName] = saslSsl
+		configMap[saslMechanismPropertyName] = plain
+		configMap[saslJaasConfigPropertyName] = getJaasValue(remoteApiKey, remoteApiSecret)
+	}
+	localApiKey, err := cmd.Flags().GetString(localApiKeyFlagName)
+	if err != nil {
+		return err
+	}
+	localApiSecret, err := cmd.Flags().GetString(localApiSecretFlagName)
+	if err != nil {
+		return err
+	}
+	if localApiKey != "" || localApiSecret != "" {
```

Comment:
why is this `||` while the equivalent for remote above is `&&`?

Reply:
Hmm I think it should be `&&` here. Will fix.

Reply:
Fixed.